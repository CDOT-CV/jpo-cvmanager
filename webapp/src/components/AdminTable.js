import React from "react";
import MaterialTable from "@material-table/core";
import { ThemeProvider, createTheme } from "@mui/material";

import "../components/css/Admin.css";

const AdminTable = (props) => {
  const defaultMaterialTheme = createTheme({
    palette: {
      common: {
        black: '#000000',
        white: '#ffffff',
      },
      primary: {
        main: '#d16d15',
        light: '#0e2052',
        contrastTextColor: '#0e2052'
      },
      secondary: {
        main: '#d16d15',
        light: '#0e2052',
        contrastTextColor: '#0e2052'
      },
      text: {
        primary: "#ffffff",
        secondary: "#ffffff",
        disabled: "#ffffff",
        hint: "#ffffff"
      },
      divider: '#333',
      background:{
        paper: "#333",
        default: "#1c1d1f"
      }
    },
    components: {
      MuiIcon: {
        styleOverrides: {
          root: {
            // Match 24px = 3 * 2 + 1.125 * 16
            color: "#d16d15"
          },
        },
      },
      TextField: {
        
      }
    },
    input: {
      color: "#11ff00",
    },
  });

  return (
    <div>
      <ThemeProvider theme={defaultMaterialTheme}>
        <MaterialTable
          actions={props.actions}
          columns={props.columns}
          data={props.data}
          title={props.title}
          editable = {props.editable}
          options={{
            selection: true,
            actionsColumnIndex: -1,
            tableLayout: "fixed",
            rowStyle: {
              overflowWrap: "break-word"
            },
          }}
          
        />
      </ThemeProvider>
    </div>
  );
};
export default AdminTable;
