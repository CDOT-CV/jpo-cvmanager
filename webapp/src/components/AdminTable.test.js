import React from "react";
import { render } from "@testing-library/react";
import AdminTable from "./AdminTable";

it("should take a snapshot", () => {
  const { asFragment } = render(<AdminTable />);

  expect(asFragment(<AdminTable />)).toMatchSnapshot();
});
