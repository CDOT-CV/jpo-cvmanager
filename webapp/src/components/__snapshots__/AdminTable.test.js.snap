// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<div>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 css-q89nbr-MuiPaper-root"
      style="position: relative;"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-ig9rso-MuiToolbar-root"
      >
        <div
          class="MuiBox-root css-i6bazn"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
            style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
          >
            Table Title
          </h6>
        </div>
        <div
          class="MuiBox-root css-15k1nh5"
        />
        <div
          class="MuiFormControl-root MuiTextField-root css-28z0on-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd css-7a9xrk-MuiInputBase-root-MuiInput-root"
          >
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
              <svg
                aria-hidden="true"
                aria-label="Search"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                data-mui-internal-clone-element="true"
                data-testid="search"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </div>
            <input
              aria-label="Search"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-ih4o70-MuiInputBase-input-MuiInput-input"
              id="mui-2"
              placeholder="Search"
              type="text"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
            >
              <button
                aria-label="Clear Search"
                class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  aria-label="clear"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                  data-testid="clear"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-1smue61"
        >
          <div>
            <div
              style="display: flex;"
            >
              <span>
                <div
                  style="display: flex;"
                />
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-187halg"
      >
        <div>
          <div
            style="overflow-y: auto;"
          >
            <div>
              <table
                class="MuiTable-root css-lenezg-MuiTable-root"
                style="table-layout: fixed;"
              >
                <thead
                  class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
                >
                  <tr
                    class="MuiTableRow-root MuiTableRow-head css-1btqj0-MuiTableRow-root"
                  >
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-paddingNone MuiTableCell-sizeMedium css-lbs2hf-MuiTableCell-root"
                      scope="col"
                      style="width: 48px;"
                    >
                      <span
                        class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-bc9mg9-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          type="checkbox"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CheckBoxOutlineBlankIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
                >
                  <tr
                    class="MuiTableRow-root css-1g6nw5f-MuiTableRow-root"
                    style="height: 245px;"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mocked-MuiTableCell-root"
                      colspan="1"
                      style="padding-top: 0px; padding-bottom: 0px; text-align: center;"
                    >
                      No records to display
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
      <table
        class="MuiTable-root css-lenezg-MuiTable-root"
      >
        <tfoot
          class="MuiTableFooter-root css-9cgc89-MuiTableFooter-root"
          style="display: grid;"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-footer css-1g6nw5f-MuiTableRow-root"
            style="display: grid;"
          >
            <td
              class="MuiTableCell-root MuiTableCell-footer MuiTableCell-sizeMedium MuiTablePagination-root css-mocked-MuiTableCell-root-MuiTablePagination-root"
              colspan="3"
            >
              <div
                class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
              >
                <div
                  class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
                />
                <p
                  class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
                  id="mui-8"
                >
                  Rows per page:
                </p>
                <div
                  class="MuiInputBase-root MuiInputBase-colorPrimary css-1ifxpwq-MuiInputBase-root-MuiTablePagination-select"
                >
                  <div
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-labelledby="mui-8 mui-7"
                    class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-1di3egx-MuiSelect-select-MuiInputBase-input"
                    id="mui-7"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="MuiBox-root css-c8mw91"
                    >
                      5 rows 
                    </div>
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                    tabindex="-1"
                    value="5"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </div>
                <p
                  class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
                >
                  0–0 of 0
                </p>
                <div
                  class="MuiBox-root css-15z8oqz"
                >
                  <span
                    aria-label="First Page"
                    class=""
                    data-mui-internal-clone-element="true"
                  >
                    <button
                      aria-label="First Page"
                      class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="first_page"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M18.41 16.59 13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
                        />
                      </svg>
                    </button>
                  </span>
                  <span
                    aria-label="Previous Page"
                    class=""
                    data-mui-internal-clone-element="true"
                  >
                    <button
                      aria-label="Previous Page"
                      class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="chevron_left"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                        />
                      </svg>
                    </button>
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
                    style="flex: 1; text-align: center; align-self: center; flex-basis: inherit;"
                  >
                    0-0 of 0
                  </span>
                  <span
                    aria-label="Next Page"
                    class=""
                    data-mui-internal-clone-element="true"
                  >
                    <button
                      aria-label="Next Page"
                      class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="chevron_right"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                        />
                      </svg>
                    </button>
                  </span>
                  <span
                    aria-label="Last Page"
                    class=""
                    data-mui-internal-clone-element="true"
                  >
                    <button
                      aria-label="Last Page"
                      class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="last_page"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M5.59 7.41 10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
                        />
                      </svg>
                    </button>
                  </span>
                </div>
              </div>
            </td>
          </tr>
        </tfoot>
      </table>
    </div>
  </div>
</div>
`;
